.github/workflows/live-service-testing.yml

```diff
@@ -25,11 +25,6 @@ on:
 env:
   DANDI_API_KEY: ${{ secrets.DANDI_API_KEY }}
 
-# We are adding concurrency here to avoid concurrent writes to the dandi API.
```

Comment:
I thought that we needed to cancel this within the same PR:

```
concurrency:  # Cancel previous workflows on the same pull request
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true
```

Reply:
No, bc it should be handled by the deploy-tests's concurrency settings

Reply:
See, that's my confusing. I thought that that was for each of the jobs on the matrix and that this top level thing would contro inter-PR.

Reply:
I think this section prevents jobs within the matrix from running at the same time (macos-13 python 3.9, windows  python 3.12, etc.), which is doing most of the work to prevent a race condition.
```yaml
strategy:
      # This avoids two tests within the matrix running at the same time
      max-parallel: 1
```

Reply:
The concurrency was trying to prevent different PRs from racing, but it was interfering with the concurrency defined at the deploy-tests.yml level.

Reply:
That's my theory at any rate.